services/utils/apiRequests.js

```diff
@@ -0,0 +1,32 @@
+import { clientEnv } from "@config/schemas/clientSchema";
+
+export async function sendRequest(requestConfig) {
+  const url = `${clientEnv.NEXT_PUBLIC_BASE_URL}${requestConfig.url}`;
+  try {
+    const res = await fetch(url, {
+      method: requestConfig.method ? requestConfig.method : "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: requestConfig.body ? JSON.stringify(requestConfig.body) : null,
+    });
+    const data = await res.json();
+    if (!res.ok) {
+      throw data.message || data.error || res.statusText;
+    }
+    return data;
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+}
+
+export function formatErrorMsg(err) {
+  let errorMsg = err;
```

Comment:
Why do we need to set errorMsg here? 

Because if the error is not an object, it's returned anyway. 

So perhaps just have if (typeof err === "object") and then return the result of Object.keys.

Else return error.

Also for readability, it's sometimes better to not shorten the variable and just set the variable to "error" instead of "err"

Reply:
Yes, I agree with you👍